### PR TITLE
Added link to "go shopping" button

### LINKED ISSUE #3625: Onboarding - customizing appearance - homepage template: "go shopping" button doesn't work
**Describe the bug**
Nothing happens when I click on the "Go shopping" button that we display on the homepage that's created during the onboarding. I'm having the same problem with the 2 homepage templates: single product and multiple products.

**To Reproduce**
1. Create a store with the new onboarding
2. Visit the Customize Appearance task in the task list
3. Click to proceed with the install of sample products
4. Click to create a homepage
5. Visit the homepage page editor
6. Preview the homepage and click in the "go shopping button"

**Expected behavior**
It should redirect the user to the shop page

**Screenshots**
https://d.pr/i/3etuFu

### PR DESCRIPTION
**Pull request by octaedro**
Fixes #3625

Added a link to the shop page in the "go shopping" button

### Screenshots
screenshot-shop-page-link2

### Detailed test instructions:
1. Create a store with the new onboarding
2. Visit the Customize Appearance task in the task list
3. Click to proceed with the install of sample products
4. Click to create a homepage
5. Visit the homepage page editor
6. Preview the homepage and click in the `go shopping` button
7. The shop link page will be visible

### CODE REVIEW

File: src/API/OnboardingTasks.php [hunk 1]
```diff
@@ -185,6 +185,7 @@ public static function sanitize_special_column_name_regex( $value ) {
 	 * @return string Block content.
 	 */
 	private static function get_homepage_cover_block( $image ) {
+		$shop_url = site_url( '/?post_type=product' );
```
Comment (jeffstieler) [inline src/API/OnboardingTasks.php]: What do you think about using core's helper methods to retrieve this?

E.g.: `$shop_url = get_permalink( woocommerce_get_page_id( 'shop' ) );`

_See: https://www.skyverge.com/blog/get-woocommerce-page-urls/_

Comment (octaedro) [reply src/API/OnboardingTasks.php]: Sounds much better!

File: src/API/OnboardingTasks.php [hunk 2]
```diff
@@ -196,7 +197,7 @@ private static function get_homepage_cover_block( $image ) {
 			<!-- /wp:paragraph -->
 
 			<!-- wp:button {"align":"center"} -->
-			<div class="wp-block-button aligncenter"><a class="wp-block-button__link">' . __( 'Go shopping', 'woocommerce-admin' ) . '</a></div>
+			<div class="wp-block-button aligncenter"><a href="' . $shop_url . '" class="wp-block-button__link">' . __( 'Go shopping', 'woocommerce-admin' ) . '</a></div>
```
Comment (jeffstieler) [inline src/API/OnboardingTasks.php]: These should use `esc_url()`. See the block cover above for an example.

**Comment by jeffstieler** [comment]
🛳 it once CI passes. 👍

**Pull request merged**